.github/workflows/cd-dev.yaml

```diff
@@ -38,6 +38,40 @@ jobs:
       chart_ver: 0.1.0-dev.${{ needs.compute-sha.outputs.sha_short }}
     secrets: inherit
 
+  limit-deploy:
+    name: SSH and Limit Deployments
+    runs-on: ubuntu-latest
+    steps:
+      - name: SSH and Check Deployments
+        uses: appleboy/ssh-action@v1.2.0
+        with:
+          host: ${{ secrets.SSH_HOST }}
+          username: ${{ secrets.SSH_USERNAME }}
+          key: ${{ secrets.SSH_KEY }}
+          script: |
+            set -e  # Exit immediately if a command fails
+
+            # Get bt-dev-app- deployments sorted by creation timestamp
+            deployments=$(kubectl get deployments \
+              --namespace=bt \
+              --sort-by='.metadata.creationTimestamp' \
+              -o custom-columns=":metadata.name,:metadata.creationTimestamp" | grep '^bt-dev-app') || true
+            deployment_count=$(echo "$deployments" | wc -l)
+            
+            # Check if deployment count > 16 (2 per deployment)
+            if [ "$deployment_count" -gt 16 ]; then
+              echo "Too many deployments. Deleting the oldest deployment."
+
+              # Get oldest deployment from first line of deployments
+              oldest_deployment=$(echo "$deployments" | head -n 1 | grep -o '^bt-dev-app-[a-f0-9]\{7\}')
+
+              # Delete deployment
+              kubectl delete deployment "${oldest_deployment}-frontend" --namespace=bt
+              kubectl delete deployment "${oldest_deployment}-backend" --namespace=bt
+            else
+              echo "Deployment count is <= 16."
+            fi
+            
```

Comment:
could you rebase onto `gql` so these changes are removed

Reply:
rebased